### PR TITLE
Authentication, user model

### PR DESCRIPTION
**Pull request by OlegMR0**

### CODE REVIEW

File: src/main/java/com/example/bookstore/dto/mapper/UserMapper.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.example.bookstore.dto.RegisterUserDto;
+import com.example.bookstore.dto.UserResponseDto;
+import com.example.bookstore.model.User;
+import org.mapstruct.Mapper;
+import org.mapstruct.NullValueCheckStrategy;
+
+@Mapper(componentModel = "spring", nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS)
+public interface UserMapper {
+
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/dto/mapper/UserMapper.java]:
```suggestion
```

File: src/main/java/com/example/bookstore/dto/mapper/UserMapper.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    User toModel(RegisterUserDto userDto);
+
+    UserResponseDto toResponseDto(RegisterUserDto registerUserDto);
+
+    UserResponseDto toResponseDto(User user);
+
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/dto/mapper/UserMapper.java]:
```suggestion
```

File: src/main/java/com/example/bookstore/dto/RegisterUserDto.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+public class RegisterUserDto {
+    @NotBlank
+    private String email;
+    @NotBlank
+    private String password;
+    private String repeatPassword;
+    private String firstName;
+    private String lastName;
+    private String shippingAddress;
+
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/dto/RegisterUserDto.java]:
```suggestion
```

File: src/main/java/com/example/bookstore/dto/RegisterUserDto.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+@FieldMatch(fields = {"password", "repeatPassword"})
+public class RegisterUserDto {
+    @NotBlank
+    private String email;
+    @NotBlank
+    private String password;
+    private String repeatPassword;
+    private String firstName;
+    private String lastName;
+    private String shippingAddress;
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/dto/RegisterUserDto.java]: if they can't be nullable, add validation here as well

File: src/main/java/com/example/bookstore/validation/FieldMatchValidator.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            }
+
+            if (fieldValues.stream().distinct().count() > 1) {
+                return false;
+            }
+            return true;
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException("Fields don't match or wrong field names!");
+        }
+
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/validation/FieldMatchValidator.java]:
```suggestion
```

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bookstore/dto/RegisterUserDto.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    @NotBlank
+    private String email;
+    @NotBlank
+    private String password;
+    private String repeatPassword;
+    @NotBlank
+    private String firstName;
+    @NotBlank
+    private String lastName;
+    private String shippingAddress;
```
Comment (Sanderfuck) [inline src/main/java/com/example/bookstore/dto/RegisterUserDto.java:19]: Too poor validation

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bookstore/dto/RegisterUserDto.java (hunk 6 above)
Comment (OlegMR0) [reply src/main/java/com/example/bookstore/dto/RegisterUserDto.java:19]: Can you explain what's wrong exactly? 
NotBlank annotation checks for null value and its trimmed length, so I can't figure out what else is required here.

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**